src/main/java/com/igot/cb/community/service/impl/CommunityManagementServiceImpl.java

```diff
@@ -1874,4 +1890,86 @@ private String validateJoinPayload(Map<String, Object> request) {
         return str.toString();
     }
 
+    public List<Object> fetchUserFromprimary(List<String> userIds) {
+        List<Object> userList = new ArrayList<>();
+        Map<String, Object> propertyMap = new HashMap<>();
+        propertyMap.put(Constants.ID, userIds);
+        List<Map<String, Object>> userInfoList = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+            Constants.KEYSPACE_SUNBIRD, Constants.TABLE_USER, propertyMap,
+            Arrays.asList(Constants.PROFILE_DETAILS, Constants.FIRST_NAME, Constants.ID), null);
+
+        userList = userInfoList.stream()
+            .map(userInfo -> {
+                Map<String, Object> userMap = new HashMap<>();
+
+                // Extract user ID and user name
+                String userId = (String) userInfo.get(Constants.ID);
+                String userName = (String) userInfo.get(Constants.FIRST_NAME);
+
+                userMap.put(Constants.USER_ID_KEY, userId);
+                userMap.put(Constants.FIRST_NAME_KEY, userName);
+
+                // Process profile details if present
+                String profileDetails = (String) userInfo.get(Constants.PROFILE_DETAILS);
+                if (StringUtils.isNotBlank(profileDetails)) {
+                    try {
+                        // Convert JSON profile details to a Map
+                        Map<String, Object> profileDetailsMap = objectMapper.readValue(
+                            profileDetails,
+                            new TypeReference<HashMap<String, Object>>() {
+                            });
+                        userMap.put(Constants.PROFILE_IMG_KEY, "");
+                        userMap.put(Constants.DESIGNATION_KEY, "");
+                        userMap.put(Constants.DEPARTMENT, "");
+                        userMap.put(Constants.PROFILE_STATUS, "");
+
+                        // Check for profile image and add to userMap if available
+                        if (MapUtils.isNotEmpty(profileDetailsMap)) {
+                            if (profileDetailsMap.containsKey(Constants.PROFILE_IMG)
+                                && StringUtils.isNotBlank(
+                                (String) profileDetailsMap.get(Constants.PROFILE_IMG))) {
+                                userMap.put(Constants.PROFILE_IMG_KEY,
+                                    (String) profileDetailsMap.get(Constants.PROFILE_IMG));
+                            }
+                            if (profileDetailsMap.containsKey(Constants.DESIGNATION_KEY)
```

Comment:
designation would be available inside professionalDetails array object... cross check this once again.